docs/overview/handling_resources.md

```diff
@@ -71,6 +71,23 @@ val groupedFileData: IO[IOException, Unit] =
 
 Like `ensuring`, `acquireReleaseWith` has compositional semantics, so if one `acquireReleaseWith` is nested inside another `acquireReleaseWith`, and the outer resource is acquired, then the outer release will always be called, even if, for example, the inner release fails.
 
+For resources which implement the AutoClosable interface, the convenience method `fromAutoClosable` can be used, which can be seen as the ZIO equivalent of try-with-resource.
+
+```scala mdoc:invisible
+import zio._
+import java.io.FileInputStream
+def openFileInputStream(name: String): IO[Throwable, FileInputStream] = ZIO.attemptBlocking(new FileInputStream(name))
+```
+
+```scala mdoc:silent
+val bytesInFile: IO[Throwable, Int]      = {
+  for {
+    stream <- ZIO.scoped(ZIO.fromAutoCloseable(openFileInputStream("data.json")))
```

Comment:
This is not correct because `ZIO.scoped` closes the resource when the specified workflow completes execution so the `FileInputStream` will be closed when you attempt to use it in the line below. Instead you should move `scoped` around the entire for comprehension.

Reply:
Ah, good to know. I'll fix it.